### ISSUE TITLE
Allowing for certain layers to be called depending on the input

### ISSUE THREAD

**Issue by dhrushilbadani** [keras-team/keras#13349]
I'm trying to develop a model which picks the best candidate out of a list of candidates:

- Input: A list of `K` candidates
- Target output: A one-hot vector `x` with `x[t] = p_t` where `p_t` is the probability of `t` being the best candidate

This is how I'm trying to do it in Keras:

 1. For a given training row, represent each candidate `i` as a feature vector `f_i` (all features are positive). The row may have less than `K` candidates; in that case, I mask
    / pad up the list of candidates with dummy `f_i` vectors (of all
    -1's)
 2. Pass each of the actual `f_i` vectors through a set of layers to get some score `s_i`; the dummy feature vectors shouldn't be passed
    through these layers and should directly get a score of 0
 3. Do
    softmax over all the non-zero `s_i` and get a final probability

Step 1 is trivial, and step 3 can be done with a custom activation as shown [here][1]. Step 2 is what I don't know how to do; I tried using a `Lambda` layer from Keras with `K.switch` like:

```
score_layer = Lambda(lambda x: K.switch(K.sum(x) < 0, 0.0, score_using_layers(x))
```

but this didn't work. When I tried to compile the model and print the summary, it wouldn't even show the layers being used to score the real feature vectors. Can Keras add a layer for this?

  [1]: https://stackoverflow.com/questions/56078515/keras-masking-zero-before-softmax
